rclpy/rclpy/impl/implementation_singleton.py

```diff
@@ -27,6 +27,15 @@
 """
 
 import importlib
-
-
-rclpy_implementation = importlib.import_module('._rclpy', package='rclpy')
+import os
+import warnings
+
+try:
+    rclpy_implementation = importlib.import_module('._rclpy', package='rclpy')
+except ImportError as e:
+    if os.path.isfile(e.path):
+        warnings.warn(
+            "The C extension '%s' failed to be imported while being present on the system."
+            " Please refer to '%s' for possible solutions\n" %
```

Comment:
Is the trailing `\n` necessary?

Reply:
It was not necessary but made the custom message not visible at all originally that's why I added it. Not that it's part of the exception message it's not necessary anymore